src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityAssemblyLine.java

```diff
@@ -60,4 +74,128 @@ public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart) {
+                    beamCount = currentBeamCount;
+                    writeCustomData(GregtechDataCodes.UPDATE_PARTICLE, this::writeParticles);
+                }
+            }
+            else if (beamCount != 0) {
+                beamCount = 0;
+                writeCustomData(GregtechDataCodes.UPDATE_PARTICLE, this::writeParticles);
+            }
+        }
+    }
+
+    @Override
+    public void writeInitialSyncData(PacketBuffer buf) {
+        super.writeInitialSyncData(buf);
+        writeParticles(buf);
+    }
+
+    @Override
+    public void receiveInitialSyncData(PacketBuffer buf) {
+        super.receiveInitialSyncData(buf);
+        readParticles(buf);
+    }
+
+    @Override
+    public void receiveCustomData(int dataId, PacketBuffer buf) {
+        if (dataId == GregtechDataCodes.UPDATE_PARTICLE) {
+            readParticles(buf);
+        } else {
+            super.receiveCustomData(dataId, buf);
+        }
+    }
+
+    @Override
+    public void onRemoval() {
+        super.onRemoval();
+        if (getWorld().isRemote && beamParticles != null) {
+            for (GTLaserBeamParticle[] particle : beamParticles) {
+                if (particle[0] != null) {
+                    particle[0].setExpired();
+                    particle[1].setExpired();
+                }
+            }
+        }
+        beamParticles = null;
+    }
+
+    private void writeParticles(PacketBuffer buf) {
+        buf.writeVarInt(beamCount);
+    }
+
+    private void readParticles(PacketBuffer buf) {
+        beamCount = buf.readVarInt();
+        if (beamParticles == null) {
+            beamParticles = new GTLaserBeamParticle[17][2];
+        }
+        BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos(getPos());
+        for (int i = 0; i < beamParticles.length; i++) {
+            GTLaserBeamParticle particle = beamParticles[i][0];
+            if (i < beamCount && particle == null) {
+                pos.setPos(getPos());
+                Vector3 startPos = new Vector3().add(
+                                pos.move(getFrontFacing().rotateY().getOpposite(), i))
+                        .add(0.5, 0, 0.5);
+                Vector3 endPos = startPos.copy().subtract(0, 1, 0);
+
+                beamParticles[i][0] = new GTLaserBeamParticle(getWorld(), startPos, endPos)
+                        .setBody(laserLocation)
+                        .setBeamHeight(0.125f)
+                        // Try commenting or adjusting on the next four lines to see what happens
+                        .setDoubleVertical(true)
+                        .setHead(laserHeadLocation)
+                        .setHeadWidth(0.1f)
+                        .setEmit(0.2f);
+                beamParticles[i][0].setOnUpdate(p -> { // remove it if machine is inValid
+                    if (!isValid() || getWorld().getTileEntity(getPos()) != this.getHolder()) p.setExpired();
+                });
+                pos.setPos(getPos());
```

Comment:
The code which creates the new beam particle could be extracted into another method which inputs the start and end pos, and returns the new particle (to then be stored in the array after). This would reduce the duplicated code in this method - assuming that my comments about the differences between the particles are unintentional.